### ISSUE TITLE
Secure Websocket closes immediately 

### ISSUE THREAD

**Issue by marshallwalker** [ktorio/ktor#623]
The ws works flawlessly but the wss method gets closed right away.
I'm using 0.9.5

```
kotlinx.coroutines.experimental.channels.ClosedReceiveChannelException: Channel was closed
	at kotlinx.coroutines.experimental.channels.Closed.getReceiveException(AbstractChannel.kt:1067)
	at kotlinx.coroutines.experimental.channels.AbstractChannel$ReceiveElement.resumeReceiveClosed(AbstractChannel.kt:907)
	at kotlinx.coroutines.experimental.channels.AbstractSendChannel.helpClose(AbstractChannel.kt:317)
	at kotlinx.coroutines.experimental.channels.AbstractSendChannel.close(AbstractChannel.kt:254)
	at kotlinx.coroutines.experimental.channels.ChannelCoroutine.close(ChannelCoroutine.kt)
	at kotlinx.coroutines.experimental.channels.SendChannel$DefaultImpls.close$default(Channel.kt:84)
	at io.ktor.network.tls.TLSClientHandshake$input$1.doResume(TLSClientHandshake.kt:96)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resumeWithException(CoroutineImpl.kt:48)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resumeWithException(CoroutineImpl.kt:47)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:41)
	at kotlinx.coroutines.experimental.DispatchedTask$DefaultImpls.run(Dispatched.kt:149)
	at kotlinx.coroutines.experimental.io.internal.MutableDelegateContinuation.run(MutableDelegateContinuation.kt:14)
	at io.ktor.network.util.IOCoroutineDispatcher$IODispatchedTask.run(IOCoroutineDispatcher.kt)
	at io.ktor.network.util.IOCoroutineDispatcher$IOThread$run$1.doResume(IOCoroutineDispatcher.kt:73)
	at kotlin.coroutines.experimental.jvm.internal.CoroutineImpl.resume(CoroutineImpl.kt:42)
	at kotlinx.coroutines.experimental.DispatchedTask$DefaultImpls.run(Dispatched.kt:149)
	at kotlinx.coroutines.experimental.DispatchedContinuation.run(Dispatched.kt:13)
	at kotlinx.coroutines.experimental.EventLoopBase.processNextEvent(EventLoop.kt:140)
	at kotlinx.coroutines.experimental.BlockingCoroutine.joinBlocking(Builders.kt:70)
	at kotlinx.coroutines.experimental.BuildersKt__BuildersKt.runBlocking(Builders.kt:46)
	at kotlinx.coroutines.experimental.BuildersKt.runBlocking(Unknown Source)
	at io.ktor.network.util.IOCoroutineDispatcher$IOThread.run(IOCoroutineDispatcher.kt:68)
```

```kotlin
httpClient.ws(host = "echo.websocket.org") {
  send(Frame.Text("Hello World"))

  for (message in incoming.map { it as? Frame.Text }.filterNotNull()) {
    println(message.readText())
  }
}

httpClient.wss(host = "echo.websocket.org") {
  send(Frame.Text("Hello World"))

  for (message in incoming.map { it as? Frame.Text }.filterNotNull()) {
    println(message.readText())
  }
}
```

**Comment by e5l** [comment]
Hi, @marshallwalker. Thanks for the report.
It's obviously a bug.
Could you report the JVM version and OS?

**Comment by marshallwalker** [comment]
Hey @e5l Java 1.8.0_151 Windows 10

**Comment by marshallwalker** [comment]
It would seem the issue is just that Ktor WSS doesn't support very many encryption protocols.

**Comment by e5l** [comment]
The fix is done and would be available in `1.1.2`.

**Comment by NurseyitTursunkulov** [comment]
the same thing happens when I connect from android. The simple websocket closes immediately
